### PR TITLE
Update versions, set minimum maven version

### PR DESCRIPTION
**Pull request by nicktindall**
This is probably a little bit risky, I looked for the plugins that had major version bumps to see if our invocations needed to be updated but there's every chance I missed something.

### CODE REVIEW

File: java-parent-pom/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>net.openhft</groupId>
         <artifactId>root-parent-pom</artifactId>
-        <version>1.2.5</version>
+        <version>1.2.5-SNAPSHOT</version>
```
Comment (nicktindall) [inline java-parent-pom/pom.xml:27]: Strange that this one was set to 1.2.4, that hasn't been released yet (I checked in Nexus)

File: java-parent-pom/pom.xml [hunk 2]
```diff
@@ -39,7 +39,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.0.0-M4</version>
```
Comment (nicktindall) [inline java-parent-pom/pom.xml:42]: Rely on the version from the parent

File: pom.xml [hunk 3]
```diff
@@ -83,4 +83,32 @@
...
+                        -->
+                        <id>enforce-maven</id>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireMavenVersion>
+                                    <version>3.5.4</version>
+                                </requireMavenVersion>
```
Comment (nicktindall) [inline pom.xml:106]: 3.5.4 is OLD (from 2018) but I think that's the version that's on the build nodes.

Comment (nicktindall) [reply pom.xml:106]: When we're feeling brave we can try to jump to 3.8.x

File: third-party-bom/pom.xml [hunk 4]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>net.openhft</groupId>
         <artifactId>root-parent-pom</artifactId>
-        <version>1.2.4</version>
+        <version>1.2.5-SNAPSHOT</version>
```
Comment (nicktindall) [inline third-party-bom/pom.xml:27]: Again, 1.2.4 doesn't seem to be released in nexus. Not sure why its depended on here.

File: pom.xml (hunk 3 above)
Comment (minborg) [reply pom.xml:106]: At least this provides a decent baseline to start with

File: root-parent-pom/pom.xml [hunk 5]
```diff
@@ -98,13 +117,13 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>3.0.0-M4</version>
+                    <version>3.0.0-M5</version>
```
Comment (minborg) [inline root-parent-pom/pom.xml:120]: There was some kind of thread problem with one of the Ms I recall.

**Review by minborg — APPROVED** [review]
Looks good!

**Pull request merged**